modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java

```diff
@@ -1046,28 +1046,19 @@ private List<Map<String, Object>> _getNestedFields(
 			List<Map<String, Object>> nestedFields =
 				(List<Map<String, Object>>)field.get("nestedFields");
 
-			if (nestedFields != null) {
-				Stream<Map<String, Object>> stream = nestedFields.stream();
-
-				return stream.flatMap(
-					this::_getNestedFieldsStream
-				).collect(
-					Collectors.toList()
-				);
+			if (nestedFields == null) {
+				return new ArrayList<>();
 			}
 
-			return new ArrayList<>();
-		}
-
-		private Stream<Map<String, Object>> _getNestedFieldsStream(
-			Map<String, Object> field) {
+			List<Map<String, Object>> nestedFieldsList = new ArrayList<>();
 
-			List<Map<String, Object>> nestedFieldsList = new ArrayList<>(
-				Arrays.asList(field));
+			for (Map<String, Object> nestedField : nestedFields) {
+				Collections.addAll(nestedFieldsList, nestedField);
```

Comment:
@JanisZhang Please just use `list.add` here.

Reply:
Got it. Thanks!